dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,795 @@
+                SqliteConstants.VectorSearchExtensionName;
+
+            List<SqliteColumn> vectorTableColumns = SqliteVectorStoreRecordPropertyMapping.GetColumns(
+                this._vectorTableProperties.Value,
+                this._propertyReader.StoragePropertyNamesMap);
+
+            tasks.Add(this.CreateVirtualTableAsync(
+                this._vectorTableName,
+                vectorTableColumns,
+                ifNotExists,
+                extensionName!,
+                cancellationToken));
+        }
+
+        return Task.WhenAll(tasks);
+    }
+
+    private Task<int> CreateTableAsync(string tableName, List<SqliteColumn> columns, bool ifNotExists, CancellationToken cancellationToken)
+    {
+        const string OperationName = "CreateTable";
+
+        using var command = this._commandBuilder.BuildCreateTableCommand(tableName, columns, ifNotExists);
+
+        return this.RunOperationAsync(OperationName, () => command.ExecuteNonQueryAsync(cancellationToken));
+    }
+
+    private Task<int> CreateVirtualTableAsync(string tableName, List<SqliteColumn> columns, bool ifNotExists, string extensionName, CancellationToken cancellationToken)
+    {
+        const string OperationName = "CreateVirtualTable";
+
+        using var command = this._commandBuilder.BuildCreateVirtualTableCommand(tableName, columns, ifNotExists, extensionName);
+
+        return this.RunOperationAsync(OperationName, () => command.ExecuteNonQueryAsync(cancellationToken));
+    }
+
+    private Task<int> DropTableAsync(string tableName, CancellationToken cancellationToken)
+    {
+        const string OperationName = "DropTable";
+
+        using var command = this._commandBuilder.BuildDropTableCommand(tableName);
+
+        return this.RunOperationAsync(OperationName, () => command.ExecuteNonQueryAsync(cancellationToken));
+    }
+
+    private async Task<TRecord?> InternalGetAsync<TKey>(
+        TKey key,
+        GetRecordOptions? options,
+        CancellationToken cancellationToken)
+    {
+        Verify.NotNull(key);
+
+        var condition = new SqliteWhereEqualsCondition(
+            this._propertyReader.KeyPropertyStoragePropertyName,
+            key,
+            this._dataTableName);
+
+        return await this.InternalGetBatchAsync([key], condition, options, cancellationToken)
+            .FirstOrDefaultAsync(cancellationToken)
+            .ConfigureAwait(false);
+    }
+
+    private async IAsyncEnumerable<TRecord> InternalGetBatchAsync<TKey>(
+        IEnumerable<TKey> keys,
+        GetRecordOptions? options,
+        [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        var condition = new SqliteWhereInCondition(
+            this._propertyReader.KeyPropertyStoragePropertyName,
+            keys.Cast<object>().ToList(),
+            this._dataTableName);
+
+        await foreach (var record in this.InternalGetBatchAsync(keys, condition, options, cancellationToken).ConfigureAwait(false))
```

Comment:
Since this call has the same return type as the method, no need to foreach here, we can just return directly.